core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -137,6 +132,8 @@ private[spark] abstract class Task[T](
           memoryManager.synchronized { memoryManager.notifyAll() }
         }
       } finally {
+        // though we unset the ThreadLocal here, the context itself is still queried directly
```

Comment:
nit: "the context member variable" instead of just "the context" (took me a min to parse this)